core/src/main/java/org/springframework/security/oauth2/server/authorization/OAuth2AuthorizationService.java

```diff
@@ -42,4 +43,19 @@
 	 */
 	OAuth2Authorization findByTokenAndTokenType(String token, TokenType tokenType);
 
+	/**
+	 * Returns the {@link OAuth2Authorization} containing the provided {@code token},
+	 * or {@code null} if not found.
+	 *
+	 * @param token the token credential
+	 * @return the {@link OAuth2Authorization} if found, otherwise {@code null}
+	 */
+	OAuth2Authorization findByToken(String token);
```

Comment:
I'm wondering if we need to add this operation? Instead we could consider changing `findByTokenAndTokenType` to `String token, @Nullable TokenType tokenType`? I think the name would change to. Thoughts?

Reply:
Just looking up for access_token for now